core/blockchain.go

```diff
@@ -723,10 +723,16 @@ func (bc *BlockChain) Rollback(chain []common.Hash) {
 }
 
 // SetReceiptsData computes all the non-consensus fields of the receipts
-func SetReceiptsData(config *params.ChainConfig, block *types.Block, receipts types.Receipts) {
+func SetReceiptsData(config *params.ChainConfig, block *types.Block, receipts types.Receipts) error {
 	signer := types.MakeSigner(config, block.Number())
 
 	transactions, logIndex := block.Transactions(), uint(0)
+	if len(transactions) != len(receipts) {
+		log.Error("Length of transactions in block does not match with the length of receipts", "len(transaction)",
+			len(transactions), "len(receipts)", len(receipts))
+		return fmt.Errorf("length of transactions in block does not match with the length of receipts: len(transaction) is %d, len(receipts) is %d",
+			len(transactions), len(receipts))
+	}
```

Comment:
Please just `return errors.New("transaction and receipt count mismatch")` , also no need for an explicit error log.

Reply:
Done